hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/MorphingPortableReader.java

```diff
@@ -47,7 +46,7 @@ public int readInt(String fieldName) throws IOException {
             case SHORT:
                 return super.readShort(fieldName);
             default:
-                throw new IncompatibleClassChangeError();
+                throw new IncompatibleClassChangeError("This type is incompatible to read from :" + fd.getType());
```

Comment:
We don't have class name there. But we can include classId of the portable and other detail you have mentioned